packages/google_maps_flutter/google_maps_flutter_android/example/integration_test/google_maps_tests.dart

```diff
@@ -936,26 +936,26 @@ void googleMapsTests() {
     final ExampleGoogleMapController controller =
         await controllerCompleter.future;
 
+    await tester.pumpAndSettle();
+
+    // TODO(mossmana): Adding this delay addresses
+    // https://github.com/flutter/flutter/issues/131783. It may be related
+    // to https://github.com/flutter/flutter/issues/54758 and should be
+    // re-evaluated when that issue is fixed.
+    await Future<void>.delayed(const Duration(seconds: 1));
```

Comment:
this violates our style guide: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#never-check-if-a-port-is-available-before-using-it-never-add-timeouts-and-other-race-conditions

Reply:
@Hixie I entirely agree, which is why I posted the following comment https://github.com/flutter/flutter/issues/54758#issuecomment-1684226391. It is unfortunately an existing pattern in this particular test and I would like to fix it the proper way, but I believe there's additional prerequisite work to be done first.

Reply:
I tried the suggestion and `onCameraIdle` is never called.

Reply:
@stuartmorgan I documented what I tried in https://docs.google.com/document/d/1BPQ4OdlNkNEq5xUt8cp72F4qrZrOVTja4vIxqIvq3X4/edit?usp=sharing. Any other suggestions?